pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -686,6 +694,39 @@ func checkForPodSecondaryNetworkAttachment(pod *corev1.Pod) (string, bool) {
 	return netObj, netObjExists && netObj != ""
 }
 
+func (pc *PodController) isLocalPod(namespace, name string) bool {
+	_, err := pc.podLister.Pods(namespace).Get(name)
+	return err == nil
+}
+
+func (pc *PodController) localContainerIDs() sets.Set[string] {
+	validContainerIDs := sets.Set[string]{}
+	storedSecondaryInterfaces := pc.interfaceStore.GetInterfacesByType(interfacestore.ContainerInterface)
+	for _, interfaceConfig := range storedSecondaryInterfaces {
+		validContainerIDs.Insert(interfaceConfig.ContainerID)
+	}
+	return validContainerIDs
+}
+
+func (pc *PodController) cleanupStaleAddresses() {
```

Comment:
Could you add some comments to explain what case we are covering here - agent restarts, primary OVS ports in ovsdb are gone, Pods recreated with a different ContainerID?